app/assets/javascripts/covid19/src/webapp/components/structures-table/cells/EntityCell.tsx

```diff
@@ -1,21 +1,31 @@
-import React from "react";
+import React, { useState, useEffect, useRef } from "react";
 import i18n from "../../../../utils/i18n";
 import { CellProps } from "../Columns";
 import { Link } from "../Link";
+import { rowHeight } from "../StructuresTable";
+import styled from "styled-components";
+import { BadgeDetails } from "../BadgeDetails";
 
 export const EntityCell: React.FC<CellProps> = React.memo(props => {
-    const { row } = props;
+    const { row, onClickDetails, moreDetails } = props;
+    const [height, setHeight] = useState(0);
+    const ref = useRef<null | HTMLUListElement>(null);
+
+    useEffect(() => {
+        setHeight(ref.current?.getBoundingClientRect().height ?? 0);
+    }, []);
```

Comment:
All this code is repeated 4 times on different cells, right? This asks for abstraction. 2 approaches: 1) create a custom hook, 2) use a wrapper component (and use children). The best approach depends on the context.